lib/rubocop/cop/rails/arel_star.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop prevents usage of `"*"` on an Arel::Table column reference.
+      #
+      # Using `arel_table["*"]` causes the outputted string to be a literal
+      # quoted asterisk (e.g. <tt>`my_model`.`*`</tt>). This causes the
+      # database to look for a column named <tt>`*`</tt> (or `"*"`) as opposed
+      # to expanding the column list as one would likely expect.
+      #
+      # @example
+      #   # bad
+      #   MyTable.arel_table["*"]
+      #
+      #   # good
+      #   MyTable.arel_table[Arel.star]
+      #
+      class ArelStar < Cop
+        MSG = 'Use `Arel.star` instead of `"*"` for expanded column lists.'
+
+        RESTRICT_ON_SEND = %i[[]].freeze
+
+        def_node_matcher :star_bracket?, <<~PATTERN
+          (send {const {_ :arel_table}} :[] $(str "*"))
```

Comment:
`{_ :arel_table}` means "anything or `:arel_table`", which will always match.

Similarly, the entire `{const {_ :arel_table}}` expression matches any node, and is equivalent to `_`.

I guess perhaps you meant to write:

```suggestion
          (send {const (send _ :arel_table)} :[] $(str "*"))
```

...which would resolve the false positive case mentioned in https://github.com/rubocop-hq/rubocop-rails/pull/344#discussion_r497568709.

Reply:
@eugeneius This is a good change, thank you!